logic/transaction_pool.js

```diff
@@ -1020,4 +1050,46 @@ __private.expireTransactions = function(transactions, cb) {
 	);
 };
 
+/**
+ * Undo unconfirmed transaction from mem account and removes transactions from the pool if it is expired.
+ *
+ * @private
+ * @param {Object[]} transactions - Array of unconfirmed transactions to be undo and expire
+ * @param {function} cb - Callback function
+ * @returns {SetImmediate} error, ids[]
+ */
+__private.expireAndUndoUnconfirmedTransactions = (transactions, cb) => {
+	async.eachSeries(
+		transactions,
+		(transaction, eachSeriesCb) => {
+			if (!transaction) {
+				return setImmediate(eachSeriesCb);
+			}
+
+			if (__private.isExpired(transaction)) {
+				modules.transactions.undoUnconfirmed(transaction, undoUnconfirmErr => {
+					if (undoUnconfirmErr) {
+						library.logger.error(
+							`Failed to undo unconfirmed transaction: ${transaction.id}`,
+							undoUnconfirmErr
+						);
+						return setImmediate(eachSeriesCb);
+					}
+					// Remove transaction from unconfirmed, queued and multisignature lists
+					self.removeUnconfirmedTransaction(transaction.id);
+					library.logger.info(
+						`Expired transaction: ${
+							transaction.id
+						} received at: ${transaction.receivedAt.toUTCString()}`
+					);
+					return setImmediate(eachSeriesCb);
+				});
+			} else {
+				return setImmediate(eachSeriesCb);
+			}
+		},
+		() => setImmediate(cb)
```

Comment:
No error handling on this block.

Reply:
the error is handled in specific functions and logged, the reason for not propagation the error in case of `expireTransactions` is that `nextExpiry` is a job and the error doesn't lead to any action, so we just log the error and do not return the error anywhere, so let me know if we want to change to behaviour to something else.